env0/resource_template_test.go

```diff
@@ -625,6 +626,11 @@ func TestUnitTemplateResource(t *testing.T) {
 				HelmChartName:        templateUseCase.template.HelmChartName,
 			}
 
+			if templateUseCase.template.Type != "terraform" && templateUseCase.template.Type != "terragrunt" {
+				templateCreatePayload.TerraformVersion = ""
+				updateTemplateCreateTemplate.TerraformVersion = ""
```

Comment:
Just out of curiosity as I'm not too familiar with Go, `templateCreatePayload` & `updateTemplateCreateTemplate` have many common fields, is it not common to extract them to something like BaseTemplate struct? Like in Python?

Reply:
To add on this: Golang does not have inheritance and is not object oriented.
https://go.dev/doc/faq#:~:text=Types-,Is%20Go%20an%20object%2Doriented%20language%3F,in%20some%20ways%20more%20general.